src/smt/theory_str_noodler/formula_preprocess.cpp

```diff
@@ -1240,4 +1254,29 @@ namespace smt::noodler {
         }
     }
 
+    /**
+     * @brief Check if the instance is clearly unsatisfiable.
+     * 
+     * @return True --> unsat for sure.
+     */
```

Comment:
Shouldn't we put these in header files?

Reply:
Also maybe add a note that it checks trivial (dis)equations.

Reply:
1) see my comment below. Maybe we could discuss it.